### ISSUE TITLE
Caching RxJava2 call adapter

### ISSUE THREAD

**Issue by osipxd** [square/retrofit#3272]
What kind of issue is this?

 - [x] Feature Request. Start by telling us what problem you’re trying to solve. Often a solution
       already exists! Don’t send pull requests to implement new features without first getting our
       support. Sometimes we leave features out on purpose to keep the project small.
---
I use RxJava2 call adapter and faced with a problem. Invocation of API method == new call to API, even if the previous same request has not yet completed. But sometimes I want to call one API method from several UI elements that don't know about each other. And if the first call is not completed before the second call start, I can't cache its result yet to prevent the second call.  
In other words, I want to subscribe to the same Rx chain from several places if the chain is not terminated.

I've created wrapper on top of RxJava2CallAdapter that solves the problem: https://gist.github.com/osipxd/6317aefb8b01390cf870eb5bcf4c6320  
If you want I can make a pull request with this functionality to adapter-rxjava2.

**Comment by osipxd** [comment]
I think it will be clearer with the diagram.

![rx-cache-diagram](https://user-images.githubusercontent.com/3684355/72248121-71321500-3607-11ea-9595-d289f5719ef4.png)

**Comment by JakeWharton** [comment]
This is not the desired behavior of the built-in adapter.

If you want to cache in-flight requests and multicast them to new subscribers you can apply that as an operator after the call to Retrofit or use a custom `CallAdapter` to compose that functionality on top of the built-in RxJava support ([similar to this example](https://github.com/square/retrofit/blob/fdd6aa73c6d393f986eeaf04aabf68ff8b0e4840/samples/src/main/java/com/example/retrofit/RxJavaObserveOnMainThread.java)).

**Comment by osipxd** [comment]
Ok. Thank you
